### PR TITLE
make public sharing links readable

### LINKED ISSUE #239: Readable links for public calendars
I shared a public link to our [Open Source Design](http://opensourcedesign.net) calendar at https://github.com/opensourcedesign/events/issues/49#issuecomment-267208684 and as you can see the link looks pretty strange:
https://cloud.nextcloud.com/apps/calendar/public/MIFAFLFJADIVX63I

Much nicer would be:
https://cloud.nextcloud.com/apps/calendar/public/opensourcedesign

It would be great if for calendars shared via link there’s a possibility to make the link look good like that. @nextcloud/calendar

### PR DESCRIPTION
**Pull request by georgehrke**
fixes #239 

This will create links like `http://nextcloud.dev/index.php/apps/calendar/p/WX9UN5XVXXHZE444/Geburtstage-von-Freunden`

### CODE REVIEW

**Comment by jancborchardt** [comment]
Awesome! @eppfel @LukasReschke what do you think?

**Comment by georgehrke** [comment]
ping @eppfel @LukasReschke ;)

File: js/app/models/calendarListItemModel.js [hunk 1]
```diff
@@ -49,7 +49,16 @@ app.factory('CalendarListItem', function($rootScope, $window, Calendar, WebCal,
 			},
 			publicSharingURL: {
 				get: () => {
-					return $rootScope.root + 'p/' + context.calendar.publicToken;
+					let displayname = context.calendar.displayname
+						.replace(/\s+/g, '-').replace(/[^\w\-]+/g, '')
+						.replace(/\-\-+/g, '-').replace(/^-+/, '')
+						.replace(/-+$/, '');
```
Comment (eppfel) [inline js/app/models/calendarListItemModel.js]: @georgehrke Just out of curiosity. Couldn't you combine at least the combine the regex for '-' and '' with groups?

Comment (georgehrke) [inline js/app/models/calendarListItemModel.js]: Hm. maybe.
But this regex just works and i don't like touching it ;)

**Pull request merged**